pkg/skaffold/watch/triggers.go

```diff
@@ -109,3 +114,51 @@ func (t *manualTrigger) Start() (<-chan bool, func()) {
 
 	return trigger, func() {}
 }
+
+// notifyTrigger watches for changes when fsnotify
+type fsNotifyTrigger struct {
+}
+
+// Debounce tells the watcher to not debounce rapid sequence of changes.
+func (t *fsNotifyTrigger) Debounce() bool {
+	return false
+}
+
+func (t *fsNotifyTrigger) WatchForChanges(out io.Writer) {
+	color.Yellow.Fprintln(out, "FS Notify starting")
+}
+
+// Start starts listening to pressed keys.
+func (t *fsNotifyTrigger) Start() (<-chan bool, func()) {
+	trigger := make(chan bool)
+	c := make(chan notify.EventInfo, 1)
+
+	if err := notify.Watch("./...", c, notify.All); err != nil {
+		logrus.Fatalf("Unable to start notification: %v", err)
+	}
+
+	go func() {
+		for {
+			ei := <-c
+			if isMatchingFile(filepath.Base(ei.Path())) {
+				logrus.Infof("Triggering rebuild")
+				trigger <- true
+			}
+		}
+	}()
+	return trigger, func() {
+		notify.Stop(c)
+	}
+}
+
+func isMatchingFile(event string) bool {
```

Comment:
This should not be trigger's job to ignore files.

Reply:
Removed is as per our discussion